src/lib/evil/evil_dirname.c

```diff
@@ -0,0 +1,16 @@
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif /* HAVE_CONFIG_H */
+
+#include <evil_api.h>
+
+#include <stdlib.h>
+
+#include <evil_windows.h>
+
+EVIL_API char* dirname (char* path)
+{
+  static char dir[MAX_PATH];
```

Comment:
Use a threa local storage, otherwise this can't be called from multiple threads.

Reply:
Oh then we should fix evil's `basename()` too.